### ISSUE TITLE
Create internal link to an anchor ('Get Headings' button) does not work

### ISSUE THREAD

**Issue by pjdebruin64** [turnermm/ckgedit#616]
To create an internal link to a section header (anchor), one has to click the '**Get Headings**' button on the 'Advanced' tab on the 'Link' modal window.

But then the Headings dropdown menu is not filled with headings at all, but rather with the following PHP error message (HTML markup converted):

> **Warning**: require_once(C:\inetpub\wwwroot\wiki/inc/Input.class.php): failed to open stream: No such file or directory in **C:\inetpub\wwwroot\wiki\lib\plugins\ckgedit\ckeditor\get_headers.php** on line **9**
> 
> **Fatal error**: require_once(): Failed opening required 'C:\inetpub\wwwroot\wiki/inc/Input.class.php' (include_path='.;C:\php\pear') in **C:\inetpub\wwwroot\wiki\lib\plugins\ckgedit\ckeditor\get_headers.php** on line **9**
> 

Dokuwiki release 2020-07-29 "Hogfather"
ckgedit version: current_21-Feb_23-12_34
Running on IIS 7
PHP Version 7.3.1

Dokuwiki release "Hogfather" does not have a file inc/Input.class.php indeed.

**Comment by pjdebruin64** [comment]
Hello Myron, this seems not to fix the issue. Now I get:

> **Warning**:  file(C:\inetpub\data\dokuwiki/pages/C/\inetpub\data\dokuwiki/pages/datumberekeningen.txt): failed to open stream: No such file or directory in **C:\inetpub\wwwroot\Dev\dokuwiki\lib\plugins\ckgedit\ckeditor\get_headers.php** on line **33**
> 
> **Warning**:  Invalid argument supplied for foreach() in **C:\inetpub\wwwroot\Dev\dokuwiki\lib\plugins\ckgedit\ckeditor\get_headers.php** on line **35**

and still no headings in the dropdown.

**Comment by pjdebruin64** [comment]
NB "C:\inetpub\data\dokuwiki' is my savedir.

**Comment by turnermm** [comment]
Ckgedit needs relative paths, relative to the root directory of your dokuwiki.  Your setup is  as follows:

```
        dokuwiki[root]
              lib
                 plugins
                     ckgedit
                       ckeditor:  get_headers.php
```

`./data` is the default savedir.  A save directory immediately outside of the root would be `../data` and one two directories outside would be .`./../data`.  I've spent a good deal of time over the week-end working on alternate save directories.  That is a different problem  from your original.

**Comment by pjdebruin64** [comment]
Strange, both dokuwiki and ckgedit work fine using an absolute savedir (with backslashes).
'Get Headings' is the only thing that doesn't work.
I changed the savedir to a relative path ('..\..\..\data\dokuwiki') but now the 'Browse server' button on the 'Link' window doesn't work anymore, it doesn't return any output.

BTW I had enabled 'userewrite' and 'useslash'. Disabling did not help.
BTW2 I must use backslashes in 'savedir'; with forward slashes plugin 'AcMenu' doesn't work.
